.github/workflows/ci.yaml

```diff
@@ -1,33 +1,39 @@
-# This workflow will install Python dependencies, run tests and lint with a single version of Python
-# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
-
-name: Python application
+name:
 
 on:
   push:
-    branches: [ "main" ]
+    branches: ["main"]
   pull_request:
-    branches: [ "main" ]
-
-permissions:
-  contents: read
+    branches: ["main"]
```

Comment:
why not trigger the ci for all branches?

Did you turn off the permission of direct push to main branch?

Reply:
![image](https://github.com/user-attachments/assets/7005e8d2-ca67-46b3-8905-11eece8f559e)
direct push is disabled

Reply:
will turn ci on only when targeting `main` for now